src/nomad_simulations/schema_packages/numerical_settings.py

```diff
@@ -890,15 +911,295 @@ def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
         super().normalize(archive, logger)
 
 
-class BasisSet(NumericalSettings):
-    """"""
-
-    # TODO work on this base section (@ndaelman-hu)
+class BasisSet(ArchiveSection):
```

Comment:
As mentioned below, I'm very much for the name change, but it's better to perform it when migrating the parsers. This will prevent any compatibility issues.

Reply:
I think you are over-thinking this. When you are doing parsers, we will know that it is the new way. Otherwise, and immediately when parsing, one would realize that the naming of the legacy style is wrong.

Simply change it here and that's it. Forget about the old schema. Let it burn.

![984b7148-792c-4132-9ed6-3c6728084e98_942x646](https://github.com/user-attachments/assets/48f35656-ee99-42fb-ad4d-96118cfb9e42)

Reply:
Lol

I'm really not trying to preserve the old schema conventions. It's just that if I shift the name `BasisSet` around, it'll get confusing for when I hook up the parsers, e.g. this was the `basis_set`, but now it's a component that goes into the `basis_set` that used to be a container. Have mercy on my poor dev soul

Reply:
This is true if you take months to implement the new parsers, but you will see it is not the case.

Reply:
...
I'll be out on holidays for the next 2 weeks, so there's that.
After, I'd like to start migrating the parsers to the new entry-points system, with each their own repo. Migrating means, retaining the old parser, while also have the setup ready to add the new one. You off-handly suggested this last week, and I agree that it is best to decouple the parsers before deprecating the old ones.

Exciting is definitely on the top of the list, but I'm less sure about Abinit or Wien2k. Regardless, I'd handle the basis set migration for all of them at once. I'm not sure about the timeframe of when we'll get around to it, but we can bump up priority there.

Reply:
Please, just change it to the new naming. You are not the only one transfering parsers, and better to have the schema naming under control so the definitions don't changr in 1 or 2 months.

Furthermore, this plan you have in mind I think is not entirely correct. The idea would be to, first, take over one parser and transfer to the new schema. We do not need to put parsers in the new entry points mechanism with the old schema.

But you go on holidays now. Did you change everything you wanted? I can takr over and merge next week if you can't.

Reply:
I finished the majority, just some normalization and testing. Unless anyone is touching `numerical_settings.py`, just leave it and I will finish it once I'm back. You'll be back the week after, so then you review for the last time and we merge.